discovery/azure/azure.go

```diff
@@ -490,6 +494,7 @@ func mapFromVMScaleSetVM(vm compute.VirtualMachineScaleSetVM, scaleSetName strin
 	return virtualMachine{
 		ID:                *(vm.ID),
 		Name:              *(vm.Name),
+		ComputerName:      *(vm.VirtualMachineScaleSetVMProperties.OsProfile.ComputerName),
```

Comment:
Thanks for this. I tend to be very cautious  here and add a few nil checks. Are those fields all guaranteed to be here?

Thanks!

Reply:
thanks! addressed.

Reply:
we were not cautious enough :) https://github.com/prometheus/prometheus/issues/9382

Reply:
😢 sorry about this. I saw you mentioned `I will work on a fix ASAP.` already, let me know if I can help.